### PR TITLE
APIv2 container stats - send header once

### PR DESCRIPTION
**Pull request by baude**
when sending container stats, we should only set and send the response header once.  using approach developed by jhonce for his image build refactor.

Fixes: #7521

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
@jwhonce ptal

File: pkg/api/handlers/compat/containers_stats.go [hunk 1]
```diff
@@ -174,11 +189,13 @@ func StatsContainer(w http.ResponseWriter, r *http.Request) {
 			ID:       stats.ContainerID,
 			Networks: net,
 		}
-
-		utils.WriteJSON(w, http.StatusOK, s)
-		if flusher, ok := w.(http.Flusher); ok {
-			flusher.Flush()
+		// encode data
+		if err := enc.Encode(s); err != nil {
+			utils.Error(w, "failed stat encoding", http.StatusBadRequest, err)
```
Comment (jwhonce) [inline pkg/api/handlers/compat/containers_stats.go:194]: You have already send the StatusOK so you cannot send StatusBadRequest here.

Comment (baude) [reply pkg/api/handlers/compat/containers_stats.go:194]: what should i use @jwhonce

**Review by vrothberg — COMMENTED** [review]
I don't see the connection to #7521. I would expect a fix for #7521 to check in the loop if the context has been cancelled (i.e., connection has been closed) and return.

**Comment by vrothberg** [comment]
> I don't see the connection to #7521. I would expect a fix for #7521 to check in the loop if the context has been cancelled (i.e., connection has been closed) and return.

See [events](https://github.com/containers/podman/blob/master/pkg/api/handlers/compat/events.go#L120) for an example.

**Comment by vrothberg** [comment]
> > I don't see the connection to #7521. I would expect a fix for #7521 to check in the loop if the context has been cancelled (i.e., connection has been closed) and return.
> 
> See [events](https://github.com/containers/podman/blob/master/pkg/api/handlers/compat/events.go#L120) for an example.

I think we should borrow the idea from events ^ which takes care of both: sending the header only once _and_ checking for closed connections.

**Comment by vrothberg** [comment]
@baude, I can take over if you want to

File: pkg/api/handlers/compat/containers_stats.go (hunk 1 above)
Comment (jwhonce) [reply pkg/api/handlers/compat/containers_stats.go:194]: Most you can do is log error.

**Comment by vrothberg** [comment]
> @baude, I can take over if you want to

As I have your permission to take over issues, I'll close it and wrap up a new PR :)

**Pull request closed without merging**